### PR TITLE
reference: update buildx reference docs for v0.7.1

### PR DESCRIPTION
**Pull request by crazy-max**
Follow-up docker/buildx#762

Generate YAML docs from remote bake definition for buildx to keep track of changes.

Expected usage when release `v0.7.0` will be created on buildx repo:

```shell
BUILDX_REF=v0.7.0 make buildx-yaml
```

Output example:

```shell
BUILDX_REF=v0.7.0-rc1 make buildx-yaml
```
```text
DOCS_FORMATS=yaml docker buildx bake --set "*.output=/tmp/docs-output.pDANiXYDJt" "https://github.com/docker/buildx.git#v0.7.0-rc1" update-docs
#1 [internal] load git source https://github.com/docker/buildx.git#v0.7.0-rc1
#1 0.062 hint: Using 'master' as the name for the initial branch. This default branch name
#1 0.062 hint: is subject to change. To configure the initial branch name to use in all
#1 0.062 hint: of your new repositories, which will suppress this warning, call:
#1 0.062 hint:
#1 0.062 hint:  git config --global init.defaultBranch <name>
#1 0.062 hint:
#1 0.062 hint: Names commonly chosen instead of 'master' are 'main', 'trunk' and
#1 0.062 hint: 'development'. The just-created branch can be renamed via this command:
#1 0.062 hint:
#1 0.062 hint:  git branch -m <name>
#1 0.063 Initialized empty Git repository in /var/lib/buildkit/runc-overlayfs/snapshots/snapshots/48/fs/
#1 0.548 5c4e3fc860ce9896db1cfc09bfea2cd9e348e049       refs/tags/v0.7.0-rc1
#1 2.631 From https://github.com/docker/buildx
#1 2.631  * [new tag]         v0.7.0-rc1 -> v0.7.0-rc1
#1 DONE 3.4s

#2 [internal] load git source https://github.com/docker/buildx.git#v0.7.0-rc1
#2 0.402 5c4e3fc860ce9896db1cfc09bfea2cd9e348e049       refs/tags/v0.7.0-rc1
#2 0.585 hint: Using 'master' as the name for the initial branch. This default branch name
#2 0.585 hint: is subject to change. To configure the initial branch name to use in all
#2 0.585 hint: of your new repositories, which will suppress this warning, call:
#2 0.585 hint:
#2 0.585 hint:  git config --global init.defaultBranch <name>
#2 0.585 hint:
#2 0.585 hint: Names commonly chosen instead of 'master' are 'main', 'trunk' and
#2 0.585 hint: 'development'. The just-created branch can be renamed via this command:
#2 0.585 hint:
#2 0.585 hint:  git branch -m <name>
#2 0.587 Initialized empty Git repository in /var/lib/buildkit/runc-overlayfs/snapshots/snapshots/50/fs/.git/
#2 1.144 From /var/lib/buildkit/runc-overlayfs/snapshots/snapshots/48/fs
#2 1.144  * [new tag]         v0.7.0-rc1 -> v0.7.0-rc1
#2 1.144  * [new tag]         v0.7.0-rc1 -> v0.7.0-rc1
#2 1.839 Note: switching to 'FETCH_HEAD'.
#2 1.839
#2 1.839 You are in 'detached HEAD' state. You can look around, make experimental
#2 1.839 changes and commit them, and you can discard any commits you make in this
#2 1.839 state without impacting any branches by switching back to a branch.
#2 1.839
#2 1.839 If you want to create a new branch to retain commits you create, you may
#2 1.839 do so (now or later) by using -c with the switch command. Example:
#2 1.839
#2 1.839   git switch -c <new-branch-name>
#2 1.839
#2 1.839 Or undo this operation with:
#2 1.839
#2 1.839   git switch -
#2 1.839
#2 1.839 Turn off this advice by setting config variable advice.detachedHead to false
#2 1.839
#2 1.839 HEAD is now at 5c4e3fc Merge pull request #818 from crazy-max/fix-builder-flag
#2 DONE 1.9s

#3 resolve image config for docker.io/docker/dockerfile:1.3-labs
#3 ...

#4 [auth] docker/dockerfile:pull token for registry-1.docker.io
#4 DONE 0.0s

#3 resolve image config for docker.io/docker/dockerfile:1.3-labs
#3 DONE 1.2s

#5 docker-image://docker.io/docker/dockerfile:1.3-labs@sha256:250ce669e1aeeb5ffb892b18039c3f0801466536cb4210c8eb2638e628859bfd
#5 resolve docker.io/docker/dockerfile:1.3-labs@sha256:250ce669e1aeeb5ffb892b18039c3f0801466536cb4210c8eb2638e628859bfd
#5 resolve docker.io/docker/dockerfile:1.3-labs@sha256:250ce669e1aeeb5ffb892b18039c3f0801466536cb4210c8eb2638e628859bfd 0.0s done
#5 CACHED

#7 [internal] load metadata for docker.io/library/golang:1.17-alpine
#7 ...

#8 [auth] library/alpine:pull token for registry-1.docker.io
#8 DONE 0.0s

#9 [auth] library/golang:pull token for registry-1.docker.io
#9 DONE 0.0s

#6 [internal] load metadata for docker.io/library/alpine:latest
#6 DONE 0.9s

#7 [internal] load metadata for docker.io/library/golang:1.17-alpine
#7 DONE 1.0s

#10 [gen 1/5] FROM docker.io/library/alpine@sha256:e1c082e3d3c45cccac829840a25941e679c25d438cc8412c2fa221cf1a824e6a
#10 resolve docker.io/library/alpine@sha256:e1c082e3d3c45cccac829840a25941e679c25d438cc8412c2fa221cf1a824e6a 0.1s done
#10 DONE 0.1s

#13 [docsgen 1/3] FROM docker.io/library/golang:1.17-alpine@sha256:1ad5c2b30ee6217fb1396fd8230e3c19e6f11a5b534791a76220dd45bd3906c9
#13 resolve docker.io/library/golang:1.17-alpine@sha256:1ad5c2b30ee6217fb1396fd8230e3c19e6f11a5b534791a76220dd45bd3906c9 0.0s done
#13 sha256:66b6b52ae600093bcea2a88e347c180ddf79cf361a229c968ab134f79566115b 155B / 155B 0.1s done
#13 sha256:0d416b4d4fcca0271f31d7d73ef5910b705bc9c7e6c47e2849f526c61323bba9 0B / 110.11MB 0.2s
#13 sha256:0d416b4d4fcca0271f31d7d73ef5910b705bc9c7e6c47e2849f526c61323bba9 6.29MB / 110.11MB 0.8s
#13 sha256:0d416b4d4fcca0271f31d7d73ef5910b705bc9c7e6c47e2849f526c61323bba9 15.73MB / 110.11MB 1.5s
#13 sha256:0d416b4d4fcca0271f31d7d73ef5910b705bc9c7e6c47e2849f526c61323bba9 22.02MB / 110.11MB 2.1s
#13 sha256:0d416b4d4fcca0271f31d7d73ef5910b705bc9c7e6c47e2849f526c61323bba9 29.36MB / 110.11MB 2.7s
#13 sha256:0d416b4d4fcca0271f31d7d73ef5910b705bc9c7e6c47e2849f526c61323bba9 36.70MB / 110.11MB 3.3s
#13 sha256:0d416b4d4fcca0271f31d7d73ef5910b705bc9c7e6c47e2849f526c61323bba9 44.04MB / 110.11MB 3.9s
#13 sha256:0d416b4d4fcca0271f31d7d73ef5910b705bc9c7e6c47e2849f526c61323bba9 50.33MB / 110.11MB 4.5s
#13 sha256:0d416b4d4fcca0271f31d7d73ef5910b705bc9c7e6c47e2849f526c61323bba9 57.67MB / 110.11MB 5.1s
#13 sha256:0d416b4d4fcca0271f31d7d73ef5910b705bc9c7e6c47e2849f526c61323bba9 65.01MB / 110.11MB 5.7s
#13 sha256:0d416b4d4fcca0271f31d7d73ef5910b705bc9c7e6c47e2849f526c61323bba9 72.35MB / 110.11MB 6.3s
#13 sha256:0d416b4d4fcca0271f31d7d73ef5910b705bc9c7e6c47e2849f526c61323bba9 78.64MB / 110.11MB 6.9s
#13 sha256:0d416b4d4fcca0271f31d7d73ef5910b705bc9c7e6c47e2849f526c61323bba9 85.98MB / 110.11MB 7.5s
#13 sha256:0d416b4d4fcca0271f31d7d73ef5910b705bc9c7e6c47e2849f526c61323bba9 93.32MB / 110.11MB 8.1s
#13 sha256:0d416b4d4fcca0271f31d7d73ef5910b705bc9c7e6c47e2849f526c61323bba9 100.66MB / 110.11MB 8.7s
#13 sha256:0d416b4d4fcca0271f31d7d73ef5910b705bc9c7e6c47e2849f526c61323bba9 106.95MB / 110.11MB 9.3s
#13 sha256:0d416b4d4fcca0271f31d7d73ef5910b705bc9c7e6c47e2849f526c61323bba9 110.11MB / 110.11MB 9.6s done
#13 extracting sha256:0d416b4d4fcca0271f31d7d73ef5910b705bc9c7e6c47e2849f526c61323bba9
#13 extracting sha256:0d416b4d4fcca0271f31d7d73ef5910b705bc9c7e6c47e2849f526c61323bba9 3.7s done
#13 extracting sha256:66b6b52ae600093bcea2a88e347c180ddf79cf361a229c968ab134f79566115b 0.0s done
#13 DONE 13.3s

#14 [docsgen 2/3] WORKDIR /src
#14 DONE 0.4s

#15 [docsgen 3/3] RUN --mount=target=.   --mount=target=/root/.cache,type=cache   go build -mod=vendor -o /out/docsgen ./docs/generate.go
#15 DONE 30.5s

#11 [gen 2/5] RUN apk add --no-cache rsync git
#11 CACHED

#12 [gen 3/5] WORKDIR /src
#12 CACHED

#16 [gen 4/5] COPY --from=docsgen /out/docsgen /usr/bin
#16 DONE 0.2s

#17 [gen 5/5] RUN --mount=target=/context   --mount=target=.,type=tmpfs <<EOT (set -e...)
#17 0.572 INFO: Generating YAML for "docker buildx bake"
#17 0.573 INFO: Generating YAML for "docker buildx build"
#17 0.574 INFO: Generating YAML for "docker buildx create"
#17 0.575 INFO: Generating YAML for "docker buildx du"
#17 0.575 INFO: Generating YAML for "docker buildx imagetools create"
#17 0.576 INFO: Generating YAML for "docker buildx imagetools inspect"
#17 0.576 INFO: Generating YAML for "docker buildx imagetools"
#17 0.576 INFO: Generating YAML for "docker buildx inspect"
#17 0.576 INFO: Generating YAML for "docker buildx install"
#17 0.576 INFO: Generating YAML for "docker buildx ls"
#17 0.577 INFO: Generating YAML for "docker buildx prune"
#17 0.578 INFO: Generating YAML for "docker buildx rm"
#17 0.578 INFO: Generating YAML for "docker buildx stop"
#17 0.578 INFO: Generating YAML for "docker buildx uninstall"
#17 0.579 INFO: Generating YAML for "docker buildx use"
#17 0.579 INFO: Generating YAML for "docker buildx version"
#17 0.579 INFO: Generating YAML for "docker buildx"
#17 DONE 0.6s

#18 [update 1/1] COPY --from=gen /out /out
#18 DONE 0.0s

#19 exporting to client
#19 copying files 131.16kB 0.0s done
#19 DONE 0.0s
rm -rf ./_data/buildx/*
cp -R "/tmp/docs-output.pDANiXYDJt"/out/reference/*.yaml ./_data/buildx/
rm -rf /tmp/docs-output.pDANiXYDJt/*
```

cc @thaJeztah @tonistiigi 

Signed-off-by: CrazyMax <crazy-max@users.noreply.github.com>

### CODE REVIEW

**Comment by crazy-max** [comment]
Just tested with `BUILDX_REF=v0.7.0-rc1 make buildx-yaml` and push changes. Seems to work fine.

**Comment by crazy-max** [comment]
@thaJeztah Seems like some markdown is not correctly handled by Jekyll: https://github.com/docker/docker.github.io/runs/4114420444?check_suite_focus=true#step:6:18

```
- ./_site/engine/reference/commandline/buildx_bake/index.html
Ran on 1564 files!
  *  [go-cty](https://github.com/zclconf/go-cty/tree/main/cty/function/stdlib) is an invalid URL (line 731)

     <a href="[go-cty](https://github.com/zclconf/go-cty/tree/main/cty/function/stdlib)"> stdlib functions</a>
```

**Comment by thaJeztah** [comment]
Interesting; so it did create a link, but for some reason of the _whole_ markdown link 🤔 

Let me check the source (almost at my computer)

**Comment by thaJeztah** [comment]
Looks like the source is wrong?

    You can use [`go-cty` stdlib functions]([go-cty](https://github.com/zclconf/go-cty/tree/main/cty/function/stdlib)).

**Comment by crazy-max** [comment]
> Looks like the source is wrong?

Yes indeed, was looking at the wrong line.

File: _data/buildx/docker_buildx_build.yaml [hunk 1]
```diff
@@ -588,6 +652,27 @@ examples: |-
   $ docker buildx create --use --name insecure-builder --buildkitd-flags '--allow-insecure-entitlement security.insecure'
   $ docker buildx build --allow security.insecure .
   ```
+
+  ### Size of `/dev/shm` (--shm-size) {#shm-size}
```
Comment (thaJeztah) [inline _data/buildx/docker_buildx_build.yaml:656]: We haven't shipped buildx 0.7 yet with Docker Desktop (or the linux packages), so perhaps we should wait for the next Docker Desktop release 🤔

Comment (thaJeztah) [reply _data/buildx/docker_buildx_build.yaml:656]: Which reminds me; could you open a PR to update docker-ce-packaging? https://github.com/docker/docker-ce-packaging/blob/master/plugins/buildx.installer

Comment (crazy-max) [reply _data/buildx/docker_buildx_build.yaml:656]: Yes indeed we can keep it in draft in the meantime

**Comment by crazy-max** [comment]
Buildx 0.7.1 is now shipped with [Docker Desktop 4.3.0](https://docs.docker.com/desktop/windows/release-notes/#docker-desktop-430).

Updated with `BUILDX_REF=v0.7.1 make buildx-yaml`.

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,21 @@
+ifneq (, $(BUILDX_BIN))
+	export BUILDX_CMD = $(BUILDX_BIN)
+else ifneq (, $(shell docker buildx version))
+	export BUILDX_CMD = docker buildx
+else ifneq (, $(shell which buildx))
+	export BUILDX_CMD = $(which buildx)
```
Comment (thaJeztah) [inline Makefile:6]: nit; perhaps this should use `command -v` instead of `which` 🤔

**Review by thaJeztah — APPROVED** [review]
LGTM, thanks!

I was wondering what options we would have to integrate the `bake` flow into `docker compose` (as the repository already uses compose to create local previews; perhaps some target in compose that's using a separate `profile` could integrate it somehow).

Definitely not for this PR though; this should be good to go 👍

**Pull request merged**